src/dirac_cwl_proto/commands/__init__.py

```diff
@@ -0,0 +1,34 @@
+from __future__ import annotations
+
+from abc import ABC, abstractmethod
+from typing import List
+
+# ! This creates a circular import
+# This happens because The JobProcessor requires info from the JobWrapper,
+#  which needs info from the plugins and hints ...
+# from dirac_cwl_proto.job.job_wrapper import JobWrapper
+
+
+class CommandBase(ABC):
+    @abstractmethod
+    def execute(self, job_wrapper):
+        raise NotImplementedError()
+
+
+class JobProcessorBase:
+    preprocess_commands: List[type[CommandBase]] = []
+    postprocess_commands: List[type[CommandBase]] = []
```

Comment:
And this should be moved into https://github.com/aldbr/dirac-cwl-proto/blob/5ed828cc018d87151e407f3c4fa43d62f18cf206/src/dirac_cwl_proto/execution_hooks/core.py#L218

Note: you can see that `pre_process` and `post_process` already exist, and you can just add your logic there.